src/math.jl

```diff
@@ -2,22 +2,28 @@ Base.:*(l::Dimensions, r::Dimensions) = @map_dimensions(+, l, r)
 Base.:*(l::Quantity, r::Quantity) = Quantity(l.value * r.value, l.dimensions * r.dimensions)
 Base.:*(l::Quantity, r::Dimensions) = Quantity(l.value, l.dimensions * r)
 Base.:*(l::Dimensions, r::Quantity) = Quantity(r.value, l * r.dimensions)
-Base.:*(l::Quantity, r::Number) = Quantity(l.value * r, l.dimensions)
-Base.:*(l::Number, r::Quantity) = Quantity(l * r.value, r.dimensions)
-Base.:*(l::Dimensions, r::Number) = Quantity(r, l)
-Base.:*(l::Number, r::Dimensions) = Quantity(l, r)
+Base.:*(l::Quantity, r) = Quantity(l.value * r, l.dimensions)
+Base.:*(l, r::Quantity) = Quantity(l * r.value, r.dimensions)
+Base.:*(l::Dimensions, r) = Quantity(r, l)
+Base.:*(l, r::Dimensions) = Quantity(l, r)
```

Comment:
It seems that both Dimensions and Quantity are given many number-like methods. Might it be simpler to only do this for Quantity, and use `Quantity(true, Dimension(...))` in place of a naked Dimension?

Or some special One if `true` is too narrow. But I'm not so sure how much it's intended to support `Quantity([1,2,3], time=1)`. It would seem simplest to me to restrict Quantity to real numbers, but maybe you have other plans.

Actually, when would methods like `*(::Number, ::Dimension)` be used at all? #22 defines `const m = Quantity(1.0, length=1)`. It could instead use this method and define `m::Dimension`, but instead seems to focus on numbers.

Reply:
(Sorry, just realized all my messages were listed as pending because they are submitted as a batch when I hit "Review changes"...)